src/main/java/org/skriptlang/skript/lang/experiment/Experiment.java

```diff
@@ -0,0 +1,187 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package org.skriptlang.skript.lang.experiment;
+
+import ch.njol.skript.patterns.PatternCompiler;
+import ch.njol.skript.patterns.SkriptPattern;
+import org.jetbrains.annotations.ApiStatus;
+
+import java.util.Objects;
+
+/**
+ * An optional, potentially-experimental feature enabled per-script with the {@code using X} syntax.
+ * Experiments provided by Skript itself are found in {@link Feature}.
+ * This can also represent an unknown experiment 'used' by a script that was not declared or registered
+ * by Skript or any of its addons.
+ */
+public interface Experiment {
+
+	@ApiStatus.Internal
+	static Experiment unknown(String text) {
+		return new UnmatchedExperiment(text);
+	}
+
+	/**
+	 * A constant experiment provider (designed for the use of addons).
+	 * @param codeName The debug 'code name' of this feature.
+	 * @param phase The stability of this feature.
+	 * @param patterns What the user may write to match the feature. Defaults to the codename if not set.
+	 * @return An experiment flag.
+	 */
+	static Experiment constant(String codeName, LifeCycle phase, String... patterns) {
+		return new ConstantExperiment(codeName, phase, patterns);
+	}
+
+	/**
+	 * @return The patterns that can be written in the `using` structure.
+	 */
+	default String[] patterns() {
```

Comment:
Is this used internally anywhere? I would prefer to only require an Experiment to provide a SkriptPattern. A pattern can be converted back into a readable string if necessary. Additionally, we could rename the other method from `compiledPattern` to just `pattern`

Reply:
It was just a legacy from when this was a record before I remembered we couldn't use records. I'll remove it if you don't think it's useful.